### ISSUE TITLE
invalid_auth on some Slack workspaces

### ISSUE THREAD

**Issue by dzehme** [rusq/slackdump#40]
**Describe the bug**
Using enterprise/paid version of Slack, after configuring a secrets.txt file, running slackdump -u, it is unable to download the users or anything else:
2022/03/16 20:19:25 > checking user cache...
2022/03/16 20:19:25   caching users for the first time
2022/03/16 20:19:25 error fetching users: couldn't fetch users

verbose output from slackdump -v -u:
2022/03/16 20:22:22 slackdump.go:88: > checking user cache...
2022/03/16 20:22:22 users.go:37:   caching users for the first time
2022/03/16 20:22:22 main.go:65: error fetching users: couldn't fetch users

**To Reproduce**
When I used chrome to get the value of the authorization cookie, I did have to do URL decode so that it would appear as a base64 value.   It does not appear that it is complaining about authorization.

**Expected behavior**
Expected it to list the users

**Output**
If applicable, add output to help explain your problem.
Output is shown above.

**Desktop (please complete the following information):**
 - OS: [e.g. macOS]
 Windows 11 using released binaries.

**Additional context**

**Comment by rusq** [comment]
Hey @dzehme, this error is returned if for some reason slack API returns 0 users to a GetUsers request.  

If your goal is to fetch the users only, then there's no quick workaround until I figure out what's going on.

But if you are after a conversation or a thread, you can run it like this:
```
./slackdump -no-user-cache <URL of the conversation/thread>
```

Let me know if it works for you or you get `invalid_auth`.

If you still get `invalid_auth`, can you please run it with trace:
```
./slackdump -v -u -trace=trace.out
```
and upload the trace file to dropbox, so I could see what's going on:  https://www.dropbox.com/request/eEK5YK1srbvXPyBmnvys

Sorry for the inconvenience.

**Comment by dzehme** [comment]
So I am trying to do one conversation.   The conversation is "an archive of a deactivated account".
```
[C:\Users\dzehm\Downloads\sd]slackdump.exe -no-user-cache https://app.slack.com/client/T02EF2PP5QS/D02J8AK3161
2022/03/19 09:07:08 > checking user cache...
2022/03/19 09:07:08 error processing: "https://app.slack.com/client/T02EF2PP5QS/D02J8AK3161" (conversation will be skipped): unsuuported URL type
2022/03/19 09:07:08 job finished, dumped 0 item(s)
```
I have uploaded the trace.out for slackdump.exe -v -u -trace=trace.out command as well.   Thanks!

**Comment by rusq** [comment]
Looks like you copied the URL from the browser address bar - try right clicking the conversation and choosing "copy link" - that is supported URL.  Maybe I should implement the support for address bar urls too

**Comment by dzehme** [comment]
```
slackdump -no-user-cache https://securid.slack.com/archives/D02J8AK3161
2022/03/19 12:29:25 > checking user cache...
2022/03/19 12:29:26 error processing: "https://securid.slack.com/archives/D02J8AK3161" (conversation will be skipped): invalid_auth
2022/03/19 12:29:26 job finished, dumped 0 item(s)
```
Now I am getting the invalid_auth.   I double checked the token and cookie and they are the same values that I had before.   So I think the trace I uploaded before should still be the problem.

**Comment by rusq** [comment]
The error is the trace file is the same as you got during the last execution: "invalid_auth" which means that token and or cookie is invalid.  Slack sometimes rotates those, so:
1. can you make sure that you're using the valid token/cookie pair?
2. After you have verified that token and cookie is correct, and they match the ones that you have in secrets.txt, can you let me know if your environment use SSO or not?

Thanks

**Comment by dzehme** [comment]
So I had verified the token and cookie when I had seen the invalid_auth.   Our environment does use SSO.

**Comment by simao-silva** [comment]
I am using Linux and I've done the steps above but got the error `slack token or cookie not specified`

**Comment by rusq** [comment]
Hey @simao-silva , when specifying the cookie file, do you specify the token as well?  The cookie file does not exclude the need for the token.

**Comment by simao-silva** [comment]
@rusq I misspelled the name of the .env file. Still, I've got the error `invalid_auth: failed to authenticate:  please double check that token/cookie value are correct`

**Comment by rusq** [comment]
@simao-silva could you try the v1.3.5 release?

1. Login into slack in your browser.
2. Get the token the usual way, make sure it's the fresh one.
3. Save the cookies to disk with this [extension](https://chrome.google.com/webstore/detail/get-cookiestxt/bgaddhkoddajcdgocldbbfleckgcbcid).
4. Start slackdump using the following command:

```shell
slackdump -trace=trace.out -u -cookie slack.com_cookies.txt
```

This will output the user list to your terminal, if your credentials are correct. It will also generate a trace file.

If you still get the auth error, could you upload the trace file to the dropbox?
https://www.dropbox.com/request/eEK5YK1srbvXPyBmnvys

**Comment by simao-silva** [comment]
@rusq I am still getting `invalid_auth`. I've uploaded my trace.out to the link you provided.

**Comment by rusq** [comment]
Hey @simao-silva , i've checked the trace, and there's nothing new there - token is detected, cookies are detected and are in a file.  Maybe that has something to do with your workspace setup?  I'm not sure - I don't have many slack instances to test, and on those that I do have access - I can't reproduce this.  I will park this issue for now, maybe someone with the knowledge of Go and with the same Slack setup could lend a hand in this.

Does your instance use SSO?  If so, what provider?  I think that might help in understanding what's going on.

**Comment by dzehme** [comment]
Just wanted to let you know that I finally got a chance to try this again tonight.   My token had not changed but I commented out the cookie in my secrets.txt so it just had the token.   I then created the cookies.txt file using the extension you recommended.   Then invoked your tool with -cookie cookies.txt and the URL for the conversation I wanted.  It downloaded the entire conversation.  Thank you very much.
